src/main/java/edu/ie3/datamodel/io/processor/timeseries/TimeSeriesProcessor.java

```diff
@@ -0,0 +1,313 @@
+      WeatherValue weatherValue = (WeatherValue) entry.getValue();
+
+      Map<String, Method> irradiationFieldToMethod = extractFieldToMethod(WEATHER_IRRADIATION);
+      valueResult.putAll(processObject(weatherValue.getIrradiation(), irradiationFieldToMethod));
+
+      Map<String, Method> temperatureFieldToMethod = extractFieldToMethod(WEATHER_TEMPERATURE);
+      valueResult.putAll(processObject(weatherValue.getTemperature(), temperatureFieldToMethod));
+
+      Map<String, Method> windFieldToMethod = extractFieldToMethod(WEATHER_WIND);
+      valueResult.putAll(processObject(weatherValue.getWind(), windFieldToMethod));
+    }
+
+    /* Join all information and sort them */
+    Map<String, String> combinedResult = new HashMap<>();
+    combinedResult.putAll(timeSeriesResults);
+    combinedResult.putAll(entryResults);
+    combinedResult.putAll(valueResult);
+    return putUuidFirst(combinedResult);
+  }
+
+  /**
+   * Extracts the field name to method map for the specific source
+   *
+   * @param source Source to extract field name to methods for
+   * @return Field name to methods for the desired source
+   */
+  private Map<String, Method> extractFieldToMethod(FieldSourceToMethod.FieldSource source) {
+    return fieldToSource.entrySet().stream()
+        .filter(entry -> entry.getValue().getSource().equals(source))
+        .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().getMethod()));
+  }
+
+  @Override
+  protected Optional<String> handleProcessorSpecificQuantity(
+      Quantity<?> quantity, String fieldName) {
+    Optional<String> normalizedQuantityValue = Optional.empty();
+    switch (fieldName) {
+      case "energy":
+      case "eConsAnnual":
+      case "eStorage":
+        normalizedQuantityValue =
+            quantityValToOptionalString(quantity.asType(Energy.class).to(StandardUnits.ENERGY_IN));
+        break;
+      case "q":
+        normalizedQuantityValue =
+            quantityValToOptionalString(
+                quantity.asType(Power.class).to(StandardUnits.REACTIVE_POWER_IN));
+        break;
+      case "p":
+      case "pMax":
+      case "pOwn":
+      case "pThermal":
+        normalizedQuantityValue =
+            quantityValToOptionalString(
+                quantity.asType(Power.class).to(StandardUnits.ACTIVE_POWER_IN));
+        break;
+      default:
+        log.error(
+            "Cannot process quantity with value '{}' for field with name {} in input entity processing!",
+            quantity,
+            fieldName);
+        break;
+    }
+    return normalizedQuantityValue;
+  }
+
+  @Override
+  public String[] getHeaderElements() {
+    return flattenedHeaderElements;
+  }
+
+  @Override
+  protected List<Class<? extends TimeSeries>> getEligibleEntityClasses() {
+    return eligibleKeys.stream()
+        .map(TimeSeriesProcessorKey::getTimeSeriesClass)
+        .distinct()
+        .collect(Collectors.toList());
+  }
+}
```

Comment:
again, understand what is happening more or less but would like to get a clearer vision